versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/persist/Persist.java

```diff
@@ -120,23 +120,57 @@ Reference updateReferencePointer(@Nonnull Reference reference, @Nonnull ObjId ne
    * <p><em>Do not use this function from service implementations, use {@link ReferenceLogic}
    * instead!</em>
    *
+   * <p>This function leverages the references-cache, if enabled, and must only be used for read
+   * operations. Updating operations must use {@link #fetchReferenceForUpdate(String)}.
+   *
+   * <p>Database specific implementations of {@link Persist} must implement this function, without
+   * caching.
+   *
    * @return the reference or {@code null}, if it does not exist
    */
   @Nullable
   Reference fetchReference(@Nonnull String name);
 
+  /**
+   * This is similar to {@link #fetchReference(String)}, but does always fetch the reference from
+   * the backend, <em>refreshing</em> the references-cache.
+   *
+   * <p>Database specific implementations of {@link Persist} must not implement this function.
+   */
+  @Nullable
+  default Reference fetchReferenceForUpdate(@Nonnull String name) {
```

Comment:
"bypassingCache" would be more correct - but do we care that much?

Reply:
"ForUpdate" looks really odd in code that just tries to load the latest value :shrug: 

Why not add a parameter to the old method: `fetchReference(String name, boolean refresh)`, for example?

Reply:
That caused like 5248 additional changed lines of code ;)

Reply:
but we could have it as an overloaded method?

Reply:
I'd prefer to keep the impact low (it's would be rather cosmetic).

Reply:
Adding `fetchReference(String name, boolean refresh)` instead of `fetchReferenceForUpdate(@Nonnull String name)` is about the same in terms of impact, is it not?.. At the same time I find that `fetchReference(name, true)` is more intuitive (more precise) than `fetchReferenceForUpdate()` at the call sites. `fetchReferenceBypassingCache(name)` also works for me, but I think it will bring the term "cache" to implementation that do not care about caching, which is again a bit awkward. 

It could also be `fetchReference(String name, FetchMode mode)` with `FetchMode { LOCAL, FRESH }`... or similar. WDYT?